sale_quotation_sourcing/model/procurement.py

```diff
@@ -35,12 +35,13 @@ def make_po(self):
         res = {}
         to_propagate = self.browse()
         for procurement in self:
-
-            sale_line = (
-                procurement.sale_line_id or
-                procurement.move_dest_id.procurement_id.sale_line_id or
-                False
-            )
+            curr_proc = procurement
+            sale_line = False
+            while curr_proc:
+                if curr_proc.sale_line_id:
+                    sale_line = curr_proc.sale_line_id
+                    break
+                curr_proc = curr_proc.move_dest_id.procurement_id
```

Comment:
you can probably hand craft a loop by chaining the move_dest of a procurement to the same procurement, but this will cause the scheduler to loop, and no addon should generate such a pathological configuration.